promql/engine_test.go

```diff
@@ -1053,6 +1053,10 @@ load 1ms
 
 			res := qry.Exec(test.Context())
 			require.NoError(t, res.Err)
+			if expMat, ok := c.result.(Matrix); ok {
```

Comment:
Isn't this always a matrix?

Reply:
But the type here should always be a Matrix.

Reply:
Then why isn't the test failing? :)

Reply:
Tried without @ modifier and an instant vector selector for an instant query returns a vector and not a matrix.

Reply:
> Then why isn't the test failing? :)

Because in execEvalStmt we only sort the results of range queries.

It looks like the internal everything-is-matrix is also undone in execEvalStmt, so check is infact needed.